testing/bdd/test_ebuttd_nested_elements.py

```diff
@@ -0,0 +1,130 @@
+import xml.etree.ElementTree as ET
+from pytest_bdd import scenarios, given, when, then, parsers
+
+scenarios('features/nesting/ebuttd_nested_elements.feature')
+
+@then('the p does not have a region attribute')
+def then_p_has_no_region(test_context):
+    document = test_context['ebuttd_document']
+    tree = ET.fromstring(document.get_xml())
+    elements = tree.findall('{http://www.w3.org/ns/ttml}body/{http://www.w3.org/ns/ttml}div')
+
+    assert elements[0].get("region") == "R1"
+    assert elements[0][0].get("region") == None
+
+    assert len(elements) == 3
+
+@then('the p has been removed from the div')
+def then_p_has_been_removed(test_context):
+    document = test_context['ebuttd_document']
+    tree = ET.fromstring(document.get_xml())
+    elements = tree.findall('{http://www.w3.org/ns/ttml}body/{http://www.w3.org/ns/ttml}div')
+
+    assert elements[1].get("region") == "R2"
+    assert elements[1][0].get("region") == None
+
+    assert len(elements[1]) == 1
+
+@then('the div and p regions remain the same')
+def then_div_and_p_remain_same(test_context):
+    document = test_context['ebuttd_document']
+    tree = ET.fromstring(document.get_xml())
+    elements = tree.findall('{http://www.w3.org/ns/ttml}body/{http://www.w3.org/ns/ttml}div')
+
+    assert elements[2].get("region") == "R4"
+    assert elements[2][0].get("region") == None
+
+    assert len(elements[1]) == 1
+
+@then('divs with no p elements are removed')
+def then_it_contains_no_divs(test_context):
+    document = test_context['ebuttd_document']
+    tree = ET.fromstring(document.get_xml())
+    elements = tree.findall('{http://www.w3.org/ns/ttml}body/{http://www.w3.org/ns/ttml}div')
+    for element in elements:
+        assert len(list(element)) != 0
```

Comment:
This test should take place after the denesting? So there should be no nested divs, just top-level divs that aren't empty

Reply:
The test should do what it says independently of the context of use, so in this case if the step is to check that no `div` element has zero `p` children then it should do that, by finding all `div` elements (use the `//` operator instead of the `/` operator in the xpath) and then checking that every one has at least one `p` child.